modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/PartitionAccessImpl.java

```diff
@@ -213,4 +229,8 @@ private TxStateStorage getTxStateStorage(int partitionId) {
 
         return txStateStorage;
     }
+
+    private void addToIndexes(TableRow tableRow, RowId rowId) {
+        indexes.get().forEach(index -> index.put(tableRow, rowId));
```

Comment:
It looks like we forgot about deletions. For a deletion, on the sending side, the `ReadResult.tableRow()` will return `null`, and `ResponseEntry.rowVersions[i]' will be `null`, so `IncomingSnapshotCopier` on line 268

```
TableRow tableRow = new TableRow(entry.rowVersions().get(i).rewind());
```

should make `tableRow = null` for a null `rowVersion` buffer, so `addToIndexes()` will get a `null` `TableRow`, so it should check for null and do not add anything to an index for a null value (actually, the code in `PartitionListener` seems to do just this).

Reply:
Also, we probably need to add a scenario with a DELETE to our IT tests for this feature. Existing tests only do INSERTs.

Reply:
You're right, I'll fix it.